tests/gis_tests/geoapp/test_functions.py

```diff
@@ -424,6 +432,29 @@ def test_isempty(self):
         )
         self.assertSequenceEqual(City.objects.filter(point__isempty=True), [empty])
 
+    @skipUnlessDBFeature("has_IsEmpty_function")
+    def test_isempty_null_geometry(self):
+        null_geometry = Town.objects.create(name="Nowhere", point=None)
+        Town.objects.create(name="Somewhere", point=Point(6.825, 47.1, srid=4326))
+        if connection.ops.spatialite:
```

Comment:
I was wondering if instead of doing `if connection.ops.spatialite` we shouldn't add a new feature flag `empty_geometry_is_null` and check for that here? It feels a bit cleaner to not check for a specific backend.

@claudep @felixxm Do you have an opinion on the new flag? is it a bit overkill?

Reply:
I'm not entirely sure if this is a correct behavior, https://www.gaia-gis.it/gaia-sins/spatialite-sql-5.0.1.html. Should we use `ST_IsEmpty()` on SpatiaLite? to get the same behavior on both databases :thinking:

Reply:
SpatiaLite's IsEmpty gives us -1 if the geometry is empty/null:

```py
(Pdb) Town.objects.values("point__isempty")
<QuerySet [{'point__isempty': -1}, {'point__isempty': False}]>
```

So I think we can sidestep this whole question if we just make sure that -1 is remapped to False, probably in `IsEmpty.as_sql()`. We seem to have a similar workaround in `IsValid.as_oracle()`.

***
PostGIS returns NULL instead of 0 in this case, but that seems to be a [deviation](https://postgis.net/docs/ST_IsEmpty.html) from the spec.